src/averageAges.js

```diff
@@ -55,7 +60,24 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(child => onlyWithSon
+    ? isMale(child) && people.find(mother => child.mother === mother.name)
+    : people.find(mother => child.mother === mother.name));
```

Comment:
You should the result of `people.find(mother => child.mother === mother.name)` in the variable